tensordict/prototype/tensorclass.py

```diff
@@ -356,24 +413,52 @@ def _batch_size_setter(self, new_size: torch.Size) -> None:
 def __eq__(self, other):
     if not isinstance(other, self.__class__):
         return False
-    return self.tensordict == other.tensordict
+    keys1 = set(self.non_tensordict.keys())
+    keys2 = set(other.non_tensordict.keys())
+    if len(keys1.difference(keys2)) or len(keys1) != len(keys2):
+        raise KeyError(f"keys in {self} and {other} mismatch, got {keys1} and {keys2}")
```

Comment:
@vmoens do we want to raise in this case?

Reply:
I see now that we return false if `not isinstance(other, self.__class__)`.
shouldn't be two tensorclasses classes with different keys treated the same way?